src/js/iotjs.js

```diff
@@ -196,56 +194,41 @@
   };
 
   function prepareStackTrace(throwable) {
-    return [
-      CallSite.create('main', ['<anonymous>', 1, 7]),
-      CallSite.create('main', ['<anonymous>', 2, 7]),
-      CallSite.create('main', ['<anonymous>', 3, 7]),
-      CallSite.create('main', ['<anonymous>', 4, 7]),
-    ];
+    return makeStackTraceFromDump(throwable._frames || []);
   }
 
+  var stackPropertiesDescriptor = {
+    stack: {
+      configurable: true,
+      enumerable: false,
+      get: function () {
+        if (this._stack === undefined) {
+          process._flushParserDumpFile();
+          this._stack = makeStackTraceFromDump(this._frames || []);
+        }
+        return this._stack;
+      },
+    },
+    _stack: {
+      configurable: true,
+      writable: true,
+      enumerable: false,
+    },
+  };
+  Object.defineProperties(Error.prototype, stackPropertiesDescriptor);
+
   Error.stackTraceLimit = 1;
   Error.prepareStackTrace = prepareStackTrace;
   Error.captureStackTrace = function(throwable, terminator) {
-    Object.defineProperties(throwable, {
-      stack: {
-        configurable: true,
-        get: function () {
-          return prepareStackTrace(throwable);
-        }
-      },
-      cachedStack: {
+    var frames = process._getStackFrames(11).slice(1);
+    Object.defineProperties(throwable, Object.assign({
```

Comment:
Could we split the complex expression like this:

```js
var descriptor = Object.assign({
  _frames: {
    configurable: true,
    writable: true,
    enumerable: false,
    value: frames,
  },
}, stackPropertiesDescriptor);
Object.defineProperties(throwable, descriptor);
```

That might be more clear for maintainer :)

Reply:
BTW, about the descriptor properties, 

- `configurable` might be false is OK?
- `writable` should be false? I didn't see any place we make re-assignment on `this._frames`.

Reply:
Why user need to change this value?